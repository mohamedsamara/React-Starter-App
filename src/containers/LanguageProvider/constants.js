/*
 *
 * LanguageProvider constants
 *
 */

export const SET_LOCALE = 'src/LanguageProvider/SET_LOCALE';
