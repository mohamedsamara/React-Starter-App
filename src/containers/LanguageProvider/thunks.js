/*
 *
 * LanguageProvider thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import { handleSetLocale } from './actions';

export function setLocale(lang) {
  return function(dispatch, getState) {
    localStorage.appLang = lang;
    return dispatch(handleSetLocale(lang));
  };
}
