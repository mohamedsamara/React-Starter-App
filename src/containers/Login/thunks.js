/*
 * 
 * Login
 * 
 */


// Do this in every file where you use `fetch`
import fetch from 'cross-fetch'
import validator from "validator";
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';


import {} from './actions';


/*
export function fetch() {
  return function(dispatch, getState) {
    return fetch('', {
      method: 'GET'
    })
      .then(response => {
        return response.json();
      })
      .then(data => {
      })
      .catch(error => {});
  };
}
*/
