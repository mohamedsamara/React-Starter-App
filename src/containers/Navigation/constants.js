/*
 *
 * Navigation constants
 *
 */

export const TOGGLE_DRAWER = 'src/Navigation/TOGGLE_DRAWER';
