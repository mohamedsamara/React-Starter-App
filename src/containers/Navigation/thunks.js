/*
 *
 * Navigation
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';

import { handleToggleDrawer } from './actions';

// Toggle Drawer
export function toggleDrawer(side, open) {
  const isDrawerOpen = {};
  isDrawerOpen.side = side;
  isDrawerOpen.open = open;

  return function(dispatch, getState) {
    return dispatch(handleToggleDrawer(isDrawerOpen));
  };
}
