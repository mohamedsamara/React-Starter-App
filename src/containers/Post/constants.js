/*
 *
 * Post constants
 *
 */

export const FETCH_POSTS_REQUEST = 'src/Post/FETCH_POSTS_REQUEST';
export const FETCH_POSTS_SUCCESS = 'src/Post/FETCH_POSTS_SUCCESS';
export const HANLDE_POSTS_FAILURE = 'src/Post/HANLDE_POSTS_FAILURE';
export const DELETE_POST_SUCCESS = 'src/Post/DELETE_POST_SUCCESS';
