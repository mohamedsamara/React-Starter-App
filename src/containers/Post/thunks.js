/*
 *
 * Post thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import {
  fetchPostsRequest,
  fetchPostsSuccess,
  handlePostFailure,
  deletePostSuccess,
} from './actions';

// post failure handler
export function postFailureHandler(errorData) {
  return function(dispatch, getState) {
    return dispatch(handlePostFailure(errorData));
  };
}

// fetch posts API
export function fetchPosts() {
  return function(dispatch, getState) {
    dispatch(fetchPostsRequest(true));

    return fetch('https://jsonplaceholder.typicode.com/posts', {
      method: 'GET',
    })
      .then(response => {
        if (!response.ok) {
          throw Error(response.statusText);
        }

        return response.json();
      })
      .then(data => {
        toastr.success('Success', 'Posts fetched successfully');
        dispatch(fetchPostsSuccess(data));
        dispatch(fetchPostsRequest(false));
      })
      .catch(error => {
        const errorData = {};
        errorData.isError = true;
        errorData.errorMessage = 'Failed to fetch Posts';

        toastr.error('Error', 'Error fetching todos');

        dispatch(postFailureHandler(errorData));
      });
  };
}

// delete post API
export function deletePost(id, index) {
  return function(dispatch, getState) {
    return fetch(`https://jsonplaceholder.typicode.com/posts/${id}`, {
      method: 'DELETE',
    })
      .then(response => {
        if (!response.ok) {
          throw Error(response.statusText);
        }

        return response.json();
      })
      .then(data => {
        // check if the post is deleted then dispatch an action to remove the post from the store.
        toastr.success('Success', 'Post deleted successfully');
        dispatch(deletePostSuccess(index));
      })
      .catch(error => {
        const errorData = {};
        errorData.isError = true;
        errorData.errorMessage = 'Failed to delete post';

        toastr.error('Error', 'Error deleting post');

        dispatch(postFailureHandler(errorData));
      });
  };
}
