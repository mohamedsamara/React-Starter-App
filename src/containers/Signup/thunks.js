/*
 *
 * Signup thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';

import {} from './actions';

/*
export function fetch() {
  return function(dispatch, getState) {
    return fetch('', {
      method: 'GET'
    })
      .then(response => {
        return response.json();
      })
      .then(data => {
      })
      .catch(error => {});
  };
}
*/
