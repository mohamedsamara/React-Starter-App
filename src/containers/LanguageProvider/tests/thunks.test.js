import { setLocale } from '../thunks';
import { SET_LOCALE } from '../constants';

describe('languageProvider Thunk', () => {
  it('should dispatch correct actions when handling setLocale', () => {
    const mockDispatch = jest.fn();

    const locale = 'en';

    const expected = {
      type: SET_LOCALE,
      locale,
    };

    expect(typeof setLocale()).toEqual('function');
    setLocale('en')(mockDispatch);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
    expect(mockDispatch).toMatchSnapshot();
  });
});
