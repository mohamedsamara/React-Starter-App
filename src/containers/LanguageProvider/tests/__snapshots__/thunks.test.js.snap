// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languageProvider Thunk should dispatch correct actions when handling setLocale 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "locale": "en",
        "type": "src/LanguageProvider/SET_LOCALE",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
